Add: Expand place options

1kuemmel1 opened the pull request:
**When merged this pull request will:**
- Add a new placing option: tilt
- Changed the handling of rotating and tilting of an object
- Use ACE3 Mousehint for release action
- Remove the old addAction entry
- Adaptation of stringtable.xml
- Changed handling of object status in database (save/load)
   - use ```getPosWorld ``` and ```setPosWorld``` instead of ```getPosASL``` and ```setPosASL```
   - added some comments
   - handle ```setVectorDirAndUp``` on database save and load

**Final test:**
- [x] local
- [x] server
- [x] database handling of objects (save/load)

**Screenshots**
![placeing_new_full](https://user-images.githubusercontent.com/19553317/34043786-db1b7b34-e1a2-11e7-94bd-fd41489727f6.jpg)
![placing_new_release](https://user-images.githubusercontent.com/19553317/34043792-df76a06e-e1a2-11e7-8e18-9607c6be01ac.jpg)![placing_new_hint](https://user-images.githubusercontent.com/19553317/34043796-e0e5cf24-e1a2-11e7-9ae4-a0b2e853d8c0.jpg)

Vdauphin commented:
Hello, very good idea but you also need to make it persistent ([/core/fnc/db/save.sqf#L133](https://github.com/Vdauphin/HeartsAndMinds/blob/master_daily/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/save.sqf#L133) and [/core/fnc/db/load.sqf#L287](https://github.com/Vdauphin/HeartsAndMinds/blob/master_daily/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/db/load.sqf#L287)).

I notice you use a lot of brackets. I am following this coding guidelines for those : https://ace3mod.com/wiki/development/coding-guidelines.html#55-brackets-around-code. Can you also follow it ?
For  exemple : `[(localize "STR_BTC_HAM_LOG_PLACE_RELEASE"),""] call ace_interaction_fnc_showMouseHint; //Release`
Juste write : `[localize "STR_BTC_HAM_LOG_PLACE_RELEASE", ""] call ace_interaction_fnc_showMouseHint; //Release`

Thanks see you soon !

1kuemmel1 commented:
Hi, I've updated some thinks.

> I notice you use a lot of brackets. I am following this coding guidelines for those : https://ace3mod.com/wiki/development/coding-guidelines.html#55-brackets-around-code. Can you also follow it ?

Sorry for this. 
Did I make it correct now?

I've modified the *loadObjectStatus.sqf* and *saveObjectStatus.sqf* to handle vector settings.

Database saved:
![beforedb_save](https://user-images.githubusercontent.com/19553317/34060052-58ad234c-e1e2-11e7-8253-39ecd65a92f6.jpg)

Database loaded:
![afterdb_save](https://user-images.githubusercontent.com/19553317/34060063-6103a782-e1e2-11e7-9eb8-a52f694ca9ce.jpg)

Update first comment in PR.

Vdauphin commented:
This looks perfect ! I will review it soon. Just some questions :
- Why do you use ```getPosWorld ``` and ```setPosWorld``` instead of ```getPosASL``` and ```setPosASL``` ?
- Does the `setVectorDirAndUp` is correctly broadcasted between client on dedicate server? I noticed for C4 on suicider there was not even if the Wiki say yes :/ (https://github.com/Vdauphin/HeartsAndMinds/pull/282).
- For comment, you don't need to explain each line from my point of view but do it as you like. For coding guidelines you can refer you to this : https://ace3mod.com/wiki/development/coding-guidelines.html#54-comments-in-code

1kuemmel1 commented:
> Why do you use getPosWorld and setPosWorld instead of getPosASL and setPosASL ?

Because *PositionWorld* referenced to model center. This is better (more accurate) for ```setVectorDirAndUp```. And BIS said: *But new to Arma 3 command getPosWorld is the fastest of them all.* 😄  (see: [Code Optimisation](https://community.bistudio.com/wiki/Code_Optimisation) see: *Position World is the fastest*)

> Does the setVectorDirAndUp is correctly broadcasted between client on dedicate server? I noticed for C4 on suicider there was not even if the Wiki say yes :/ (#282).

On a locally hosted dedicated server, everything seems to be fine. I'm going to do a little test on a dedicated server soon. (Tomorrow or so).

1kuemmel1 commented:
**Update:**
After a little test on a dedicated server, I can say: it's going quite well so far.

*Saved to database:*
![savedb_dedi](https://user-images.githubusercontent.com/19553317/34078120-1cc2028e-e314-11e7-9ba6-f6e58a474877.jpg)

*After server restart (loaded from database):*
![loaddb_dedi](https://user-images.githubusercontent.com/19553317/34078125-38756098-e314-11e7-8551-b9b2361f954d.jpg)

1kuemmel1 commented:
Ahh sorry, I hit the wrong button.😕 😃

Vdauphin commented:
Good job ! looks like it works prefectly!

I forgot to ask you to increment the mission version to prevent data base conflict with previous version (`1.17`). To do so just write `1.171` in https://github.com/Vdauphin/HeartsAndMinds/blob/master_daily/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L4

1kuemmel1 commented:
> I forgot to ask you to increment the mission version to prevent data base conflict with previous version (1.17).

done

Vdauphin commented on =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/log/place.sqf:
```diff
@@ -1,44 +1,69 @@
+		detach btc_log_placing_obj;
+
+		//save to DB
+		[btc_log_placing_obj] call btc_fnc_db_saveObjectStatus;
```
@1kuemmel1  Why do you use `btc_fnc_db_saveObjectStatus` here ? You are doing nothing, right ?

1kuemmel1 replied:
Why?... actually no idea. But you be right it's useless there. Just remove it.
